Sources/OpenFeature/OpenFeatureAPI.swift

```diff
@@ -100,6 +100,9 @@ extension OpenFeatureAPI {
                         continuation.resume()
                         holder.removeAll()
                     }
+                    if $0 == .error {
+                        return
+                    }
```

Comment:
maybe something like this:
```
 if $0 == .ready || $0 == .error {
                        continuation.resume()
                        holder.removeAll()
                    }
```

Reply:
So essentially we continue the async function and clear out the holder? Why do we need to clear out the holder if the ProviderError event was emitted - we didn't get a new value for the provider?

Reply:
Got it, thanks for explaining!